typings/web-inspector.d.ts

```diff
@@ -31,28 +29,33 @@ declare global {
       /** Should use a default of 0 if not defined */
       _resourceSize?: number;
       _fromDiskCache?: boolean;
+      _fromMemoryCache?: boolean;
 
       finished: boolean;
       requestMethod: string;
       statusCode: number;
-      redirectSource?: {
-        url: string;
-      }
+      redirectSource?: {url: string;};
+      redirectDestination?: {url: string;};
+      redirects?: NetworkRequest[];
       failed?: boolean;
       localizedFailDescription?: string;
 
       _initiator: Crdp.Network.Initiator;
-      _timing: Crdp.Network.ResourceTiming;
-      _resourceType: ResourceType;
+      _timing?: Crdp.Network.ResourceTiming;
+      _resourceType?: ResourceType;
       _mimeType: string;
-      priority(): 'VeryHigh' | 'High' | 'Medium' | 'Low';
-      _responseHeaders?: {name: string, value: string}[];
+      priority(): Crdp.Network.ResourcePriority;
+      initiatorRequest(): NetworkRequest | undefined;
+      _responseHeaders?: HeaderValue[];
 
       _fetchedViaServiceWorker?: boolean;
-      _frameId: Crdp.Page.FrameId;
+      _frameId?: Crdp.Page.FrameId;
       _isLinkPreload?: boolean;
-      initiatorRequest(): NetworkRequest | null;
-      redirects?: NetworkRequest[];
+    }
```

Comment:
shouldn't we be able to get rid of this and use the class definition?

Reply:
yes, that jump seemed a bit bigger in the same go since I'm using this to typecheck the other class, but we definitely will be able to